securebanking-openbanking-uk-rs-simulator/src/main/java/com/forgerock/securebanking/openbanking/uk/rs/api/obie/payment/v3_0/domesticpayments/DomesticPaymentsApiController.java

```diff
@@ -0,0 +1,130 @@
+import javax.validation.Valid;
+import java.security.Principal;
+import java.util.Optional;
+
+import static com.forgerock.securebanking.openbanking.uk.rs.api.obie.LinksHelper.createDomesticPaymentLink;
+import static com.forgerock.securebanking.openbanking.uk.rs.converter.payment.FRSubmissionStatusConverter.toOBTransactionIndividualStatus1Code;
+import static com.forgerock.securebanking.openbanking.uk.rs.converter.payment.FRWriteDomesticConsentConverter.toOBDomestic1;
+import static com.forgerock.securebanking.openbanking.uk.rs.converter.payment.FRWriteDomesticConverter.toFRWriteDomestic;
+import static com.forgerock.securebanking.openbanking.uk.rs.persistence.document.payment.FRSubmissionStatus.PENDING;
+
+@Controller("DomesticPaymentsApiV3.0")
+@Slf4j
+public class DomesticPaymentsApiController implements DomesticPaymentsApi {
+
+    private final DomesticPaymentSubmissionRepository domesticPaymentSubmissionRepository;
+    private final PaymentSubmissionValidator paymentSubmissionValidator;
+
+    public DomesticPaymentsApiController(DomesticPaymentSubmissionRepository domesticPaymentSubmissionRepository,
+                                         PaymentSubmissionValidator paymentSubmissionValidator) {
+        this.domesticPaymentSubmissionRepository = domesticPaymentSubmissionRepository;
+        this.paymentSubmissionValidator = paymentSubmissionValidator;
+    }
+
+    @Override
+    public ResponseEntity<OBWriteDomesticResponse1> createDomesticPayments(@Valid OBWriteDomestic1 obWriteDomestic1,
+                                                                           String xFapiFinancialId,
+                                                                           String authorization,
+                                                                           String xIdempotencyKey,
+                                                                           String xJwsSignature,
+                                                                           DateTime xFapiCustomerLastLoggedTime,
+                                                                           String xFapiCustomerIpAddress,
+                                                                           String xFapiInteractionId,
+                                                                           String xCustomerUserAgent,
+                                                                           HttpServletRequest request,
+                                                                           Principal principal
+    ) throws OBErrorResponseException {
+        log.debug("Received payment submission: '{}'", obWriteDomestic1);
+
+        // TODO - before we get this far, the IG will need to:
+        //      - verify the consent status
+        //      - verify the payment details match those in the payment consent
+        //      - verify security concerns (e.g. detached JWS, access token, roles, MTLS etc.)
```

Comment:
I'll remove this TODO statement (which has been repeated in each controller) when we're confident this validation is being performed.